usaspending_api/download/filestreaming/download_generation.py

```diff
@@ -736,7 +738,22 @@ def add_data_dictionary_to_zip(working_dir, zip_file_path):
     data_dictionary_file_name = "Data_Dictionary_Crosswalk.xlsx"
     data_dictionary_file_path = os.path.join(working_dir, data_dictionary_file_name)
     data_dictionary_url = settings.DATA_DICTIONARY_DOWNLOAD_URL
-    RetrieveFileFromUri(data_dictionary_url).copy(data_dictionary_file_path)
+
+    # We are currently receiving timeouts when trying to retrieve the data dictionary during
+    # the nightly pipeline. Adding retry logic here until those timeouts are resolved
+    max_attempts = 3
```

Comment:
Would it be a good idea to move these constants to our settings? Treat them as configurable things.

Reply:
I was thinking about this, but ended up not because it seemed like a one off. Wouldn't hurt though. I'll add it

Reply:
Ended up going with my first thought. The use of these variables is limited to this file and will likely not be changing much. I did move them to the top of the file as constants though

Reply:
After some discussion over Slack, decided to move the constants into the settings file in case we want to disable this retry logic in the future without code changes